File: Python/Lv.2/5.py
//다음 큰 숫자
def solution(n):
    answer = 0
    a = 0;
    
    tmp = format(n, 'b');
    for i in tmp:
        if (i == '1'):
            a += 1;
    
    n += 1;
    
    while(1):
        b = 0;
        tmp = format(n, 'b');
        for i in tmp:
            if (i == '1'):
                b += 1;
        if(b == a):
            break;
            
        n += 1;
    
    answer = n;
    
    return answer

File: C/Lv.2/5.c
//멀리 뛰기
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long solution(int n) {
    long long answer = 0;
    long long arr[2001] = { 0 };
    
    arr[1] = 1;
    arr[2] = 2;
    
    for(int i = 3; i <= n; i++)
        arr[i] = (arr[i - 1] % 1234567 + arr[i - 2] % 1234567) % 1234567;
    
    answer = arr[n];
    
    return answer;
}
   